File: spring-backend/src/main/java/com/credigo/backend/dto/ActivityLogResponse.java
package com.credigo.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ActivityLogResponse {
    
    private Long id;
    private Integer userId;
    private String username;
    private String activity;
    private String activityType;
    private String details;
    private String ipAddress;
    private LocalDateTime createdAt;
}

File: mobile/app/src/main/java/com/sia/credigo/model/ActivityLogResponse.kt
package com.sia.credigo.model

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class ActivityLogResponse(
    val id: Long,
    val userId: Int,
    val username: String,
    val activity: String,
    val activityType: String,
    val details: String?,
    val ipAddress: String?,
    @SerializedName("createdAt")
    val timestamp: String // LocalDateTime as string
)
